flow/Deque.h

```diff
@@ -42,8 +42,10 @@ class Deque {
 
 	// TODO: iterator construction, other constructors
 	Deque(Deque const& r) : arr(0), begin(0), end(r.size()), mask(r.mask) {
-		if(r.capacity() > 0)
-			arr = (T*)aligned_alloc(__alignof(T), capacity()*sizeof(T));
+		if (r.capacity() > 0) {
+			arr = (T*)aligned_alloc(std::max(__alignof(T), sizeof(void*)), capacity() * sizeof(T));
```

Comment:
I was under the impression from our [`aligned_alloc` implementation ](https://github.com/apple/foundationdb/blob/8953b196de6d68ae88b6d5c6f538da3bc0533864/flow/Platform.h#L530) that a minimum alignment of size `sizeof(void*)` was only required in some implementations. To deal with this, our macOS implementation is automatically increasing the size of an aligned request as needed.

If this is actually a problem on other platforms too, we may want to do a similar thing for them.

Reply:
Ahh, gotcha. I think (since recently) the implementation you linked is ifdef'd out, since HAS_ALIGNED_ALLOC is defined (at least on my machine locally)

Reply:
Btw the motivation for this change was so I could run ubsan on osx, I don't have any evidence of undefined behavior on linux here.

Reply:
Ah right, that change does result in a new contract for `aligned_alloc`. If our intended fix is to resolve these at the call sites, I would be in favor of removing the extra logic in our macOS `aligned_alloc` implementation to eliminate the difference and avoid confusion.

Reply:
> removing the extra logic in our macOS aligned_alloc implementation

Done